src/sentry/workflow_engine/handlers/detector/base.py

```diff
@@ -102,10 +109,27 @@ def __init__(self, detector: Detector):
         else:
             self.condition_group = None
 
-    @abc.abstractmethod
     def evaluate(
         self, data_packet: DataPacket[DataPacketType]
-    ) -> dict[DetectorGroupKey, DetectorEvaluationResult] | None:
+    ) -> dict[DetectorGroupKey, DetectorEvaluationResult]:
+        tags = {
+            "detector_type": self.detector.type,
+            "result": "unknown",
+        }
+        try:
+            value = self.evaluate_impl(data_packet)
+            tags["result"] = "tainted" if value.tainted else "success"
+            metrics.incr("workflow_engine_detector.evaluation", tags=tags, sample_rate=1.0)
+            return value.result
+        except Exception:
+            tags["result"] = "failure"
+            metrics.incr("workflow_engine_detector.evaluation", tags=tags, sample_rate=1.0)
+            raise
+
+    @abc.abstractmethod
+    def evaluate_impl(
```

Comment:
rather than changing the external interface on all of the detectors, could we just have like a `wrapped_evaluate` for our logging handlers?